### PR TITLE
Add ipadm_addrprop module

### PR DESCRIPTION
**Pull request by xen0l**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME
ipadm_addrprop

##### ANSIBLE VERSION
```
2.3
```

##### SUMMARY
Module for managing IP address properties on illumos/Solaris systems.

### CODE REVIEW

File: lib/ansible/modules/network/illumos/ipadm_addrprop.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+EXAMPLES = '''
+# Mark address on addrobj as deprecated
+ipadm_addrprop: property=deprecated value=on addrobj=e1000g0/v6
+
+# Set network prefix length for addrobj
+ipadm_addrprop: addrobj=bge0/v4 name=prefixlen value=26
+'''
+
+RETURN = '''
```
Comment (privateip) [inline lib/ansible/modules/network/illumos/ipadm_addrprop.py]: return keys must be documented

File: lib/ansible/modules/network/illumos/ipadm_addrprop.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+    if out:
+        result['stdout'] = out
+    if err:
+        result['stderr'] = err
+
+    module.exit_json(**result)
+
+
+from ansible.module_utils.basic import *
```
Comment (privateip) [inline lib/ansible/modules/network/illumos/ipadm_addrprop.py]: please use explicit imports `from ansible.module_utils.basic import AnsibleModule`

File: lib/ansible/modules/network/illumos/ipadm_addrprop.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+main()
```
Comment (privateip) [inline lib/ansible/modules/network/illumos/ipadm_addrprop.py]: please wrap in block

```
if __name__ == '__main__':
   main()
```

**Review by privateip — CHANGES_REQUESTED** [review]
* missing metadata which is required for 2.3 modules
* please consider adding unit tests

**Comment by xen0l** [comment]
@privateip I addressed you comments. As for unit tests, they will be shared among all modules to a certain degree (I am also speaking about modules, which are already included with Ansible). For that reason, I would like to add them in the next round if possible.

**Comment by xen0l** [comment]
ready_for_review

**Pull request merged**